network/src/connectivity_manager/mod.rs

```diff
@@ -376,21 +388,42 @@ where
     fn handle_request(&mut self, req: ConnectivityRequest) {
         match req {
             ConnectivityRequest::UpdateAddresses(src, address_map) => {
+                // Keep track of if any peer's addresses have actually changed, so
+                // we can log without too much spam.
+                let mut have_any_changed = false;
+
                 for (peer_id, addrs) in address_map {
-                    trace!(
-                        "Received updated addresses for peer: {} from {:?} discovery source",
-                        peer_id.short_str(),
-                        src,
-                    );
-                    // Do not include self_peer_id in the address list for dialing.
-                    if peer_id != self.self_peer_id {
-                        self.update_peer_addrs(src, peer_id, addrs);
-                        // Ensure that the next dial attempt starts from the first addr.
+                    // Do not include self_peer_id in the address list for dialing
+                    // to avoid pointless self-dials.
+                    if peer_id == self.self_peer_id {
+                        continue;
+                    }
+
+                    // Update peer's addresses
+                    let curr_addrs = self.peer_addresses.0.entry(peer_id).or_default();
+                    if curr_addrs.update(src, addrs) {
+                        // At least one peer's addresses have actually changed.
+                        have_any_changed = true;
```

Comment:
maybe we should just print the updated address? we could get a lot of noise by printing out all addresses, thoughts?

Reply:
Yeah, would be best if we could collect the id -> id changes, and print just those.

Reply:
i think i'll just print both diffs and current state. there's two orthogonal issues: 1. quickly find out what the current state is 2. trace how that state changed over time. with just the diffs, it's super annoying trying to mentally sum the diffs to find the current state.